keel-core/src/test/kotlin/com/netflix/spinnaker/keel/actuation/NewEnvironmentPromotionCheckerTests.kt

```diff
@@ -201,19 +201,24 @@ internal class NewEnvironmentPromotionCheckerTests : JUnit5Minutests {
             }
           }
 
-          test("no constraint evaluation happens") {
-            verify(exactly = 0) {
+          test("constraint evaluation happens") {
+            verify(exactly = 1) {
               environmentConstraintRunner.checkEnvironment(any())
-              environmentConstraintRunner.checkStatelessConstraints(any(), any(), any(), any())
-              repository.getQueuedConstraintApprovals(any(), any())
             }
           }
 
-          test("the artifact is approved") {
+          test("the pinned artifact is approved") {
             verify(exactly = 1) {
               repository.approveVersionFor(deliveryConfig, artifact, "1.0", environment.name)
             }
           }
+
+          test("queued constraints aren't looked at") {
+            verify(exactly = 0) {
+              environmentConstraintRunner.checkStatelessConstraints(any(), any(), any(), any())
```

Comment:
This is a very small nit, but it looks to me like this test is testing two distinct things, stateless constraints and queued constraints. I would pull out the stateless constraint check into a separate test for clarity, e.g.:

``` kotlin
          test("stateless constraints aren't looked at") {
            verify(exactly = 0) {
              environmentConstraintRunner.checkStatelessConstraints(any(), any(), any(), any())
            }
          }

```